staging/src/k8s.io/cloud-provider/cloud.go

```diff
@@ -187,6 +187,8 @@ type Instances interface {
 	InstanceExistsByProviderID(ctx context.Context, providerID string) (bool, error)
 	// InstanceShutdownByProviderID returns true if the instance is shutdown in cloudprovider
 	InstanceShutdownByProviderID(ctx context.Context, providerID string) (bool, error)
+	// InstanceMetadataByProviderID returns the instance's metadata which contains its type and nodeAddresses.
+	InstanceMetadataByProviderID(ctx context.Context, providerID, nodeName string) (*InstanceMetadata, error)
```

Comment:
I suggest removing "nodeName" from the parameters (and leave only providerID).

In case providerID is empty (or cloud provider won't return anything for it), we can return error.

[Having just providerID would make it consistent with other methods too.]

Reply:
If so, do we need get instanceMetadata by nodeName?

Reply:
Maybe - but let's maybe start with just byProviderId - I'm still not convinced it will be really needed.

Reply:
Done, looks easier.